From c329b4bea5aa79d54533a4148a8b5e7d8cdec3c6 Mon Sep 17 00:00:00 2001
From: Ian Douglas Scott <idscott@system76.com>
Date: Tue, 5 Jan 2021 08:24:44 -0800
Subject: [PATCH] Do not enforce password strength requirements

Gnome Initial Setup doesn't seem to be enforcing this, and since
`libpam-pwquality` isn't installed by default, presumably tools like
`passwd` aren't.

Perhaps the way Gnome handles this could be done better.
---
 panels/user-accounts/cc-add-user-dialog.c | 4 ++--
 panels/user-accounts/cc-password-dialog.c | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/panels/user-accounts/cc-add-user-dialog.c b/panels/user-accounts/cc-add-user-dialog.c
index 7a99b09dd..077301a38 100644
--- a/panels/user-accounts/cc-add-user-dialog.c
+++ b/panels/user-accounts/cc-add-user-dialog.c
@@ -269,10 +269,10 @@ update_password_strength (CcAddUserDialog *self)
 
         verify = gtk_entry_get_text (self->local_verify_entry);
         if (strlen (verify) == 0) {
-                gtk_widget_set_sensitive (GTK_WIDGET (self->local_verify_entry), strength_level > 1);
+                gtk_widget_set_sensitive (GTK_WIDGET (self->local_verify_entry), TRUE);
         }
 
-        return strength_level;
+        return 2;
 }
 
 static gboolean
diff --git a/panels/user-accounts/cc-password-dialog.c b/panels/user-accounts/cc-password-dialog.c
index e803ed152..aac9271fd 100644
--- a/panels/user-accounts/cc-password-dialog.c
+++ b/panels/user-accounts/cc-password-dialog.c
@@ -104,10 +104,10 @@ update_password_strength (CcPasswordDialog *self)
 
         verify = gtk_entry_get_text (self->verify_entry);
         if (strlen (verify) == 0) {
-                gtk_widget_set_sensitive (GTK_WIDGET (self->verify_entry), strength_level > 1);
+                gtk_widget_set_sensitive (GTK_WIDGET (self->verify_entry), TRUE);
         }
 
-        return strength_level;
+        return 2;
 }
 
 static void
@@ -222,7 +222,7 @@ update_sensitivity (CcPasswordDialog *self)
 
         if (self->password_mode == ACT_USER_PASSWORD_MODE_REGULAR) {
                 strength = update_password_strength (self);
-                can_change = strength > 1 && strcmp (password, verify) == 0 &&
+                can_change = strength > 1 && strlen(password) > 0 && strcmp (password, verify) == 0 &&
                              (self->old_password_ok || !gtk_widget_get_visible (GTK_WIDGET (self->old_password_entry)));
         }
         else {
-- 
2.27.0
